lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -453,6 +455,15 @@ def _extract_collection_from_git(repo_url, coll_ver, b_path):
             proc_err,
         )
 
+    has_submodules = (pathlib.Path(b_checkout_path.decode()) / '.gitmodules').is_file()
```

Comment:
`.decode()` isn't always present because this module was written at time when it still needed to support both py2 and py3. We have a set of helpers `to_native()`, `to_bytes()` and `to_text()` for dealing with this sort of conversions. It is customary to use them instead as a convention.
```suggestion
    has_submodules = (pathlib.Path(to_native(b_checkout_path)) / '.gitmodules').is_file()
```